python/grass/jupyter/interact_display.py

```diff
@@ -0,0 +1,158 @@
+            height=self.height,
+            location=center,
+            tiles="cartodbpositron",
+        )
+
+        
+    def _get_rc(self, env=None):
+        rc = {}
+        
+        if env:
+            rcfile_location = env["GISRC"]
+        else:
+            rcfile_location = os.environ["GISRC"]
+            
+        with open(rcfile_location) as f:
+            for line in f:
+                entry = line.strip("\n").split(": ")
+                rc[entry[0]] = entry[1]
+        return rc
+    
+    
+
+    def _convert_coordinates(self, x, y, proj_in):
+        """This function reprojects coordinates to WGS84, the required
+        projection for folium.
+
+        Coordinates should be a string with x and y values separated
+        by a comma.
+
+        proj_in is a proj4 string, for example, the output of g.region
+        with the `g` flag."""
+        # Reformat input
+        coordinates = f"{x}, {y}"
+        
+        # Reproject coordinates
+        coords_folium = gs.read_command(
+            "m.proj",
+            coordinates=coordinates,
+            proj_in=proj_in,
+            separator="comma",
+            flags="do",
+        )
+
+        # Reformat from string to array
+        coords_folium = coords_folium.strip()  # Remove '\n' at end of string
+        coords_folium = coords_folium.split(",")  # Split on comma
+        coords_folium = [float(value) for value in coords_folium]  # Convert to floats
+
+        return coords_folium[1], coords_folium[0]  # Return Lat and Lon
+
+    def _convert_extent(self, env=None):
+        """This function returns the bounding box of the current region
+        in WGS84, the required projection for folium"""
+
+        # Get proj of current GRASS region
+        proj = gs.read_command("g.proj", flags="jf", env=env)
+
+        # Get extent
+        extent = gs.parse_command("g.region", flags="g", env=env)
+
+        # Convert extent to EPSG:3857, required projection for Folium
+        north, east = self._convert_coordinates(extent["e"], extent["n"], proj)
+        south, west = self._convert_coordinates(extent["w"], extent["s"], proj)
+
+        extent = {'north': north, 'south': south, 'east': east, 'west': west}
+
+        return extent
+    
+    def _folium_bounding_box(self, extent):
+        """Reformats extent into bounding box to pass to folium"""
+        return [[extent['north'], extent['west']], [extent['south'], extent['east']]]
+    
+    def add_vector(self, vector):
+        """Imports vector into temporary WGS84 location, 
+        re-formats to a GeoJSON and adds to folium map"""
+        
+        # Import vector into new Location/Mapset
+        name = gs.find_file(vector, element="vector")["fullname"]
+        # Reproject vector into WGS84 Location
+        gs.run_command("v.proj",
+                       input=name,
+                       location=self._rc_original["LOCATION_NAME"],
+                       mapset=self._rc_original["MAPSET"],
```

Comment:
Maybe the mapset is not needed when v.proj gets a fully qualified name as input (since mapset is already specified in input). Read the documentation and/or test. Using the original/current mapset would be wrong (the vector is in the mapset returned by find_file, not necessarily in the current one).

I would also suggest to name the variable full_name just to be clear later what is happening or overwrite the vector variable by the result if you want shorter code (I think it would still be perfectly readable), but see my later comment about v.out.ogr. It is also somewhat confusing to name this name since one of the values find_file returns is under key `name`.